profile/profile-layered.go

```diff
@@ -253,6 +259,22 @@ func (lp *LayeredProfile) MatchFilterLists(entity *intel.Entity) (result endpoin
 	return endpoints.NoMatch, ""
 }
 
+// MatchBypassProtection checks if the entity should be denied or permitted
+// based on some bypass protection checks.
+func (lp *LayeredProfile) MatchBypassProtection(entity *intel.Entity) (endpoints.EPResult, string) {
+	if !lp.PreventBypassing() {
+		return endpoints.NoMatch, ""
+	}
+
+	// Block firefox canary domain to disable DoH
+	if strings.ToLower(entity.Domain) == "use-application-dns.net." {
+		log.Warningf("bypass protection for firefox canary")
+		return endpoints.Denied, "Firefox canary domain"
+	}
+
+	return endpoints.NoMatch, ""
+}
+
```

Comment:
I don't feel like the profiles package is the correct place to put this. How about a new file `firewall/bypassing.go` with a function similar to `DecideOnConnection` that does all the complicated prevention stuff.
This file will probably become quite a mess with all the workarounds that we will have to do.